modules/schema/src/main/java/org/apache/ignite/internal/schema/SchemaManager.java

```diff
@@ -208,15 +218,19 @@ private void setColumnMapping(SchemaDescriptor schema, int tableId) throws Execu
      * @return Schema representation.
      */
     private SchemaDescriptor loadSchemaDescriptor(int tblId, int ver) {
-        Entry entry = metastorageMgr.getLocally(schemaWithVerHistKey(tblId, ver), Long.MAX_VALUE);
+        int version = catalogService.latestCatalogVersion();
 
-        assert !entry.tombstone() : "Table " + tblId + ", version " + ver;
+        while (version >= catalogService.earliestCatalogVersion()) {
+            CatalogTableDescriptor tableDescriptor = catalogService.table(tblId, version);
 
-        byte[] value = entry.value();
+            if (tableDescriptor == null) {
+                continue;
```

Comment:
looks like `version` is not updated, causing this loop to never end... It would be nice to cover this case with test

Reply:
Good catch, thank you!

Reply:
I believe that writing a test would be problematic right now, this might be a dead code that's never executed actually, let's investigate it separately